db.go

```diff
@@ -240,6 +245,94 @@ func newDBMetrics(db *DB, r prometheus.Registerer) *dbMetrics {
 	return m
 }
 
+// DBReadOnly provides APIs for read only operations on a database.
+type DBReadOnly struct {
+	db *DB
```

Comment:
I agree with @brian-brazil and I think this is the problematic bit. I think we can reuse more if we would do vice versa. Instead of this we would have some `Querier` and `Blocker` (or `RODB`) interfaces within `DB` (: 

This has also negative effect of DBReadOnly code having an access to write methods into underlying `db`. This might cause future bugs as well.

Wonder if that is doable without any major refactors, let me experiment for a bit.

Reply:
something like `DBReader` with `Querier()` and `Block()` methods

Reply:
Alternative is not use any relation. I think this is pretty straightforward then if we need only blocks and Querier

Reply:
https://github.com/prometheus/tsdb/pull/589